### PR TITLE
Support Apple's managed app configuration

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-12133
affects: student, teacher, parent
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/AppEnvironment/AppleManagedAppConfigurationTests.swift
* Core/CoreTests/Helpers/MockAppleManagedAppConfiguration.swift
* Frameworks/CanvasKit/CanvasKit/Helpers/CKManagedAppConfiguration.h
* Frameworks/CanvasKit/CanvasKit/Helpers/CKManagedAppConfiguration.m
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student, Teacher, Parent

[MBL-12133](https://instructure.atlassian.net/browse/MBL-12133)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.95% | 93.91% | 0.04%
**Student** | 93.16% | 93.16% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.53% | 92.53% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/AppEnvironment/AppleManagedAppConfiguration.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                let enableDemo = config["enableDemo"] as? Bool,
+                enableDemo,
+                let username = config["username"] as? String,
+                let password = config["password"] as? String
+            else {
+                return nil
+            }
+            self.username = username
+            self.password = password
+            self.host = "pcraighill.instructure.com"
```
Comment (vanwagonet) [inline Core/Core/AppEnvironment/AppleManagedAppConfiguration.swift]: Should this be a secret? I know it's not the same as a password or key, but I know we used to remove hard coded domains in the old oss export script.

File: Core/Core/AppEnvironment/AppleManagedAppConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+//
+
+import Foundation
+
+/// Manages Apple MDM Configuration
+///
+/// You can test this locally with command line arguments
+// -com.apple.configuration.managed '<dict><key>enableDemo</key><true/><key>username</key><string>username</string><key>password</key><string>password</string></dict>'
+/// `username` and `password` values should match what's set in App Store Connect.
+public class AppleManagedAppConfiguration {
```
Comment (vanwagonet) [inline Core/Core/AppEnvironment/AppleManagedAppConfiguration.swift]: This seems very specific to AppStore review process, but doesn't really handle how schools could use MDM.

File: Core/Core/Login/LoginWebPresenter.swift [hunk 3]
```diff
@@ -83,7 +85,7 @@ class LoginWebPresenter {
         }
     }
 
-    func navigationActionPolicyForUrl(url: URL) -> WKNavigationActionPolicy {
+    func navigationActionPolicyForURL(url: URL) -> WKNavigationActionPolicy {
```
Comment (vanwagonet) [inline Core/Core/Login/LoginWebPresenter.swift]: 👍

File: Core/CoreTests/Login/LoginWebPresenterTests.swift [hunk 4]
```diff
@@ -175,7 +176,24 @@ class LoginWebPresenterTests: XCTestCase {
         XCTAssertEqual(resultingRequest, expectedRequest)
     }
 
-    func defuaultRequest() -> URLRequest {
```
Comment (vanwagonet) [inline Core/CoreTests/Login/LoginWebPresenterTests.swift]: sadlol.

**Comment by armstrongnate** [comment]
This relies on https://github.com/instructure/steps-canvas-ios-secrets/pull/12.

File: Frameworks/CanvasKeymaster/CanvasKeymaster.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1240,6 +1296,44 @@
...
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/TBBModal\"",
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/TPKeyboardAvoiding\"",
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/react-native-blur\"",
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/react-native-camera\"",
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/react-native-document-picker\"",
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/react-native-image-picker\"",
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/react-native-mail\"",
+					"\"${PODS_CONFIGURATION_BUILD_DIR}/yoga\"",
+					"$(PROJECT_DIR)/../../Carthage/Build/iOS",
+				);
```
Comment (vanwagonet) [inline Frameworks/CanvasKeymaster/CanvasKeymaster.xcodeproj/project.pbxproj]: Woah, what happened here?

Comment (vanwagonet) [reply Frameworks/CanvasKeymaster/CanvasKeymaster.xcodeproj/project.pbxproj]: If pods adds these, that fine.

**Pull request merged**